### PR TITLE
Replace process-markdown.js with modular system

### PR DESCRIPTION
**Pull request by ripleyaffect**
- Turn `process-markdown.js` into a set of tasks with common utilities

Also replaces functions with newer syntax for cleaner code

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "thunderbird",
```
Comment (learykara) [inline package.json]: it's a confusing thing that this is both thunderbird and eagle-flavored-thinkdown

File: src/createCurriculumManifest/process-markdown/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+const renderer = require('./renderer');
+const tasks = require('./tasks');
+
+marked.setOptions({
+  renderer,
+});
+
+module.exports = function processMarkdown(markdown) {
+  var html = marked(markdown);
+  var $ = createDom(html);
```
Comment (learykara) [inline src/createCurriculumManifest/process-markdown/index.js]: no arrow fns / `const`s?

File: src/createCurriculumManifest/process-markdown/index.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  tasks.forEach(task => task($));
```
Comment (learykara) [inline src/createCurriculumManifest/process-markdown/index.js]: what is even happening here? this function looks real clever and could maybe use some comments

File: src/createCurriculumManifest/process-markdown/tasks/changeSrcToAsset.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+const _ = require('lodash');
+
+/**
+ * In markdown, * src= attributes point to relative URLs like 'foobar.png'
+ *    This attribute is changed to asset=
+ * In the learning app, a directive prepends the S3 URL prefix and the folder
+ *    name before populating the src attribute.
+ */
+module.exports = $ => {
+  _.chain($('[src]'))
```
Comment (learykara) [inline src/createCurriculumManifest/process-markdown/tasks/changeSrcToAsset.js]: this is taking ... jQuery as an arg? I feel like something else is happening here but I can't decode it

File: src/createCurriculumManifest/process-markdown/tasks/killStyles.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+const gutil = require('gulp-util');
+
+module.exports = $ => {
+  if ($('style').length) {
+    gutil.log(gutil.colors.yellow("Style tags found...and they're gone!"));
```
Comment (learykara) [inline src/createCurriculumManifest/process-markdown/tasks/killStyles.js]: this always made me lol

File: src/createCurriculumManifest/process-markdown/tasks/setTargetAttribute_blank.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+const { getExternalLink } = require('../utils');
+
+const jqueryTarget =
+  '[href]:' +
+  'not([href*="courses.thinkful.com"]):' +
+  'not([href*="thinkful.com"])';
```
Comment (learykara) [inline src/createCurriculumManifest/process-markdown/tasks/setTargetAttribute_blank.js]: should this just search for all thinkful subdomains?

File: src/createCurriculumManifest/process-markdown/utils/getIframe.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+module.exports = (src, height = null, width = null) =>)
+  `<iframe ` +
+  `${height !== null ? `height="${height}" ` : ''} ` +
+  `${width !== null ? `width="${width}" ` : ''} ` +
```
Comment (learykara) [inline src/createCurriculumManifest/process-markdown/utils/getIframe.js]: xtra space at the end of both of these lines I think

File: src/createCurriculumManifest/process-markdown/index.js (hunk 3 above)
Comment (ripleyaffect) [reply src/createCurriculumManifest/process-markdown/index.js]: sure, each task exposes itself as a function that takes the jquery dom object and manipulates some how, this just calls all the tasks as a replacement for calling them inline

File: src/createCurriculumManifest/process-markdown/tasks/killStyles.js (hunk 5 above)
Comment (ripleyaffect) [reply src/createCurriculumManifest/process-markdown/tasks/killStyles.js]: olex's handiwork!

File: package.json (hunk 1 above)
Comment (ripleyaffect) [reply package.json]: yeah, I think it's reasonable to change it

File: src/createCurriculumManifest/process-markdown/tasks/setTargetAttribute_blank.js (hunk 6 above)
Comment (ripleyaffect) [reply src/createCurriculumManifest/process-markdown/tasks/setTargetAttribute_blank.js]: hmm, maybe? this is a restructuring and I want to keep all behavior the same for this diff

**Pull request merged**